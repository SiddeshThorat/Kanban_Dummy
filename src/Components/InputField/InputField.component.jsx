import './InputField.styles.css';
import React,{useState} from 'react';

import { uuid } from 'uuidv4';

const InputField = ({category,addTask}) => {
    const [title, setTitle] = useState("");
    
    const onInputChange = event => {
        event.preventDefault();
        setTitle(event.target.value);
    }
    const resetTitle = () => {
        setTitle("");
    }
    return(
        <div className="inputField">
            <hr />
            <input type="text"value={title} onChange={(event) => onInputChange(event)}/>
            <input type="submit" onClick={() => {
                addTask({title,category,id: uuid()})
                resetTitle()
                }}/>
        </div>
    )   
}

export default InputField;