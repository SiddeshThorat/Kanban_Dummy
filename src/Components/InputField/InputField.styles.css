
.inputField input{
    height: 50px;
    border-radius: 25px;
    margin-bottom: 5px;
}