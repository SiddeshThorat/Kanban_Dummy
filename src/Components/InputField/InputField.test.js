import React from 'react';
import { shallow } from 'enzyme'
import InputField from'./InputField.component';

describe("Testing InputField component",() => {

    it("Snapshot testing of InputField component",() => {
        const props = {
            category: "icebox",
            addTask: jest.fn()
        }
        expect(shallow(<InputField {...props}/>)).toMatchSnapshot();
    })


})


