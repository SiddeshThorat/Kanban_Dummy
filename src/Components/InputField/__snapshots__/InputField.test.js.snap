// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InputField component Snapshot testing of InputField component 1`] = `
<div
  className="inputField"
>
  <hr />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    onClick={[Function]}
    type="submit"
  />
</div>
`;
